tests/zfs-tests/tests/functional/removal/removal_with_errors.ksh

```diff
@@ -0,0 +1,86 @@
+#! /bin/ksh -p
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2014, 2017 by Delphix. All rights reserved.
+# Copyright (c) 2018 by Lawrence Livermore National Security, LLC.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/removal/removal.kshlib
+
```

Comment:
Can you add some comments here that describe on the high-level what are looking to test here, and how you go about testing it?

e.g. This test ensures that removals with errors get cancelled ...
[1] We create a pool with setup X and we do some writes on it
[2] inject some faults to ...
[3] ..etc..

Random example: https://github.com/zfsonlinux/zfs/blob/master/tests/zfs-tests/tests/functional/removal/removal_multiple_indirection.ksh